db/db.go

```diff
@@ -267,11 +267,16 @@ func IsConnectionError(err error) bool {
 		return false
 	}
 
-	// The new driver stringifies command errors as "(Name) Message" rather than just "message". Cast to the
-	// CommandError type if possible to extract the correct error message.
-	errMsg := err.Error()
-	if cmdErr, ok := err.(mongo.CommandError); ok {
-		errMsg = cmdErr.Message
+	// The new driver adds extra information to error strings. Cast to different
+	// error types if possible to extract the correct error message.
+	var errMsg string
+	switch e := err.(type) {
+	case mongo.CommandError:
+		errMsg = e.Message
+	case mongo.WriteError:
```

Comment:
it looks to me like like `WriteError`'s `.Error()` property already just returns the message. Do we need to do this? 

I wonder if a more robust solution to this whole problem is to stop matching for exact "notMaster" errors and just see if `lowerCaseError` contains `ErrNotMaster`. or could that give us false positives?

Reply:
Using strings.Contains was the first idea I brought up in Slack and I think it'll work without any false positives. We originally decided to not do this because special-casing CommandError wasn't a big deal. If we think Contains is the right way to go, it's an easy fix.

I didn't check, but I avoided using `WriteError.Error` because I like making it explicit rather than relying on the underlying Error function.